pkg/util/clusterregistry/informer.go

```diff
@@ -0,0 +1,68 @@
+package clusterregistry
+
+import (
+	"github.com/atlassian/ctrl"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	cr_v1a1 "k8s.io/cluster-registry/pkg/apis/clusterregistry/v1alpha1"
+	clientset "k8s.io/cluster-registry/pkg/client/clientset/versioned"
+	informers "k8s.io/cluster-registry/pkg/client/informers/externalversions"
+)
+
+const (
+	ByClusterLabelIndexName = "clusterLabelIndex"
+	CustomerLabel           = "customer"
+)
+
+func ByClusterLabelIndex(object interface{}) ([]string, error) {
+	obj := object.(meta_v1.Object)
+
+	clusterLabel, ok := obj.GetLabels()[CustomerLabel]
+	if !ok {
+		return nil, nil
+	}
+
+	return []string{clusterLabel}, nil
+}
+
+type ClusterInformer struct {
+	informer cache.SharedIndexInformer
+}
+
+func NewClusterInformer(config *ctrl.Config, cctx *ctrl.Context) (*ClusterInformer, error) {
```

Comment:
Ideally this package should not depend on `ctrl`. All the plumbing logic should be in the controller constructor, like in all other controllers.